### PR TITLE
feat(profiler): allow to disable memory collector

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Review by Kyle-Verhoog — DISMISSED** [review]
probably worth having a test or two to ensure the feature

File: releasenotes/notes/profiling-disable-memory-option-c6f3765b020f4bcb.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    The memory profile collector can now entirely disabled with the
```
Comment (P403n1x87) [inline releasenotes/notes/profiling-disable-memory-option-c6f3765b020f4bcb.yaml:4]:
```suggestion
    The memory profile collector can now be entirely disabled with the
```

**Review by P403n1x87 — DISMISSED** [review]

File: docs/configuration.rst [hunk 2]
```diff
@@ -230,6 +230,12 @@ below:
      - 64
      - The maximum number of frames to capture in stack execution tracing.
 
+       .. _dd-profiling-heap-enabled:
```
Comment (P403n1x87) [inline docs/configuration.rst:233]:
```suggestion
       .. _dd-profiling-memory-enabled:
```

**Review by majorgreys — DISMISSED** [review]

**Comment by jd** [comment]
Tests added @Kyle-Verhoog

**Review by P403n1x87 — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
```
/root/project/docs/configuration.rst:239:Duplicate explicit target name: "dd-profiling-heap-enabled".
```

https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/17309/workflows/cd261f89-9057-4896-96f3-def23f0821cd/jobs/1181201

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**